hazelcast/src/main/java/com/hazelcast/sql/impl/client/SqlClientUtils.java

```diff
@@ -30,15 +31,28 @@ private SqlClientUtils() {
         // No-op.
     }
 
+    // Encode the exception as SqlError
     public static SqlError exceptionToClientError(Exception exception, UUID localMemberId) {
         HazelcastSqlException sqlException = CoreQueryUtils.toPublicException(exception, localMemberId);
 
+        Throwable rootCause = ExceptionUtil.getRootCause(exception);
```

Comment:
I don't think we should return only the root cause and skip the wrapping exception. Full stacktrace should be returned, at the price of verbosity on the client (the user asked for `-verbose`, so I don't see it as an issue).

Sometimes the full stacktrace contains useful information. Similarly, we were trying to be smart with util methods like `com.hazelcast.internal.util.ExceptionUtil#peel` and it proved to be counter-productive in several cases.

Reply:
rev : e223dcc91bb014f69af8a1c3bf0a704fa22d60f1